assets/js/googlesitekit/data/create-error-store.js

```diff
@@ -92,12 +101,31 @@ export function createErrorStore() {
 
 			case CLEAR_ERROR: {
 				const { baseName, args } = payload;
-				const key = generateErrorKey( baseName, args );
+				const newState = { ...state };
+				if ( baseName ) {
+					const key = generateErrorKey( baseName, args );
+					newState.errors = { ...( state.errors || {} ) };
+					delete newState.errors[ key ];
+				} else {
+					// @TODO: remove it once all instances of the legacy behavior have been removed.
+					delete newState.error;
+				}
 
-				const errors = { ...( state.errors || {} ) };
-				delete errors[ key ];
+				return newState;
+			}
 
-				return { ...state, errors };
+			case CLEAR_ERRORS: {
+				const { baseName } = payload;
+				const newState = { ...state };
+				if ( baseName ) {
+					const key = generateErrorKey( baseName );
+					newState.errors = { ...( state.errors || {} ) };
+					delete newState.errors[ key ];
```

Comment:
This doesn't delete all errors for the `baseName`, it right now only deletes the error for `baseName` with no arguments. To actually delete all errors for `baseName`, we'll need to delete all entries in `newState.errors` where `key` is `baseName` or starts with `baseName`.

To make this bulletproof, we should probably update `generateErrorKey` to separate the `md5` bit with some distinct separator, e.g. `::`. This way we can check here as follows: `key === baseName || key.startsWith( baseName + '::' )`.